components/director/pkg/graphql/app_template_validation.go

```diff
@@ -0,0 +1,105 @@
+package graphql
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	validation "github.com/go-ozzo/ozzo-validation"
+	"github.com/kyma-incubator/compass/components/director/pkg/inputvalidation"
+	"github.com/pkg/errors"
+)
+
+func (i ApplicationTemplateInput) Validate() error {
+	return validation.Errors{
+		"rule.ValidPlaceholders": i.validPlaceholders(),
```

Comment:
Should be `rule.ValidPlaceholders` or `Rule.ValidPlaceholders`? In different files, I can see different approaches. (See PR #516)